server/container_status_linux.go

```diff
@@ -0,0 +1,29 @@
+package server
+
+import (
+	"context"
+
+	"github.com/cri-o/cri-o/internal/log"
+	"github.com/cri-o/cri-o/internal/oci"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func toCRIContainerUser(ctx context.Context, container *oci.Container) *types.ContainerUser {
+	runtimeSpec := container.Spec()
+	if runtimeSpec.Process == nil {
+		log.Infof(ctx, "Container %q has no process spec", container.ID())
+		return nil
+	}
+	user := runtimeSpec.Process.User
+	supplementalGroups := make([]int64, 0, len(user.AdditionalGids))
+	for _, gid := range user.AdditionalGids {
+		supplementalGroups = append(supplementalGroups, int64(gid))
+	}
+	return &types.ContainerUser{
```

Comment:
is this something we could store in the oci.Container object itself, rather than recalculating each time?